airflow/contrib/operators/ssh_operator.py

```diff
@@ -71,11 +71,19 @@ def execute(self, context):
             if self.ssh_conn_id:
                 if self.ssh_hook and isinstance(self.ssh_hook, SSHHook):
                     self.log.info("ssh_conn_id is ignored when ssh_hook is provided.")
+                    
+                if self.remote_host is not None:
+                    self.ssh_hook.remote_host = self.remote_host
```

Comment:
I think in this case we should ignore the remote_host parameter too.

Reply:
Not so.  This is a feature of ssh_operator that allows re-using an SSHHook (for example, across operators) and dynamically setting the remote_host.  It does have some risk in a multi-threaded execution, so... user beware.  
That said, the indentation was incorrect.  The condition applies IFF ssh_hook is defined on the ssh_operator instance.